sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_service_client.py

```diff
@@ -303,7 +303,7 @@ def undelete_file_system(self, name, version, **kwargs):
         new_name = kwargs.pop('new_name', None)
         file_system = self.get_file_system_client(new_name or name)
         self._blob_service_client.undelete_container(
-            name, version, new_name=new_name, **kwargs)  # pylint: disable=protected-access
+            name, deleted_version, new_name=new_name, **kwargs)  # pylint: disable=protected-access
```

Comment:
Should we rename FileSystemProperties.version also

Reply:
that attribute is named `.version` in undelete_container, I can change it if you feel strongly about it

Reply:
I have renamed it to `deleted_version` as well

Reply:
we shouldn't change the one in blob package, as it will become a breaking change for customer.
While do we consider to change the one in datalake package?

Reply:
It has not been changed in blob package. I'm just passing the parameter that was passed to DSC to the BSC.

Reply:
oh you only changed in datalake! Great!